### DISCUSSION TITLE
iOS 15.1.7 -[AGXA9Device originalObject]: unrecognized selector sent to instance 0x13c074c00

### DISCUSSION THREAD

**Discussion by leiyuncun** [Tencent/libpag#1948] [General]
=============异常崩溃报告=============
date:
2023-12-04 02:57:44 +0000
name:
NSInvalidArgumentException
reason:
-[AGXA9Device originalObject]: unrecognized selector sent to instance 0x13c074c00
callStackSymbols:
0   CoreFoundation                      0x0000000181253c74 717D70C9-3B8E-3ABC-AE16-050588FC3EE8 + 601204
1   libobjc.A.dylib                     0x0000000198a7fee4 objc_exception_throw + 56
2   CoreFoundation                      0x00000001813246e8 717D70C9-3B8E-3ABC-AE16-050588FC3EE8 + 1455848
3   CoreFoundation                      0x00000001811ed54c 717D70C9-3B8E-3ABC-AE16-050588FC3EE8 + 181580
4   CoreFoundation                      0x00000001811ec75c _CF_forwarding_prep_0 + 92
5   Metal                               0x000000019a61d588 _MTLCreateGLMetalDevice + 76
6   AppleMetalGLRenderer                0x00000001dcd23ce0 4B420F8E-3F8D-3864-978F-77AA65FD6CF8 + 27872
7   AppleMetalGLRenderer                0x00000001dcd1e7b4 gldCreateDevice + 72
8   libGFXShared.dylib                  0x00000001c232da2c gfxInitializeLibrary + 1920
9   GLEngine                            0x00000001c224bd1c gliInitializeLibrary + 48
10  OpenGLES                            0x00000001c231b8f8 30A206E9-035F-3FD1-A7BB-3C4254BD9D73 + 30968
11  OpenGLES                            0x00000001c231b4e0 30A206E9-035F-3FD1-A7BB-3C4254BD9D73 + 29920
12  OpenGLES                            0x00000001c231a6bc 30A206E9-035F-3FD1-A7BB-3C4254BD9D73 + 26300
13  OpenGLES                            0x00000001c231a51c 30A206E9-035F-3FD1-A7BB-3C4254BD9D73 + 25884
14  libpag                              0x000000010d42efc4 _ZNK3pag15DropShadowStyle6verifyEv + 2324
15  libpag                              0x000000010d44f498 _ZNK3pag11FillElement6verifyEv + 71004
16  libpag                              0x000000010d430a90 _ZNK3pag15DropShadowStyle6verifyEv + 9184
17  libpag                              0x000000010d465630 _ZNK3pag16BackendSemaphore6glSyncEv + 1212
18  libpag                              0x000000010d42b23c pag_display_link_set_functions + 936
19  libpag                              0x000000010d4bb5f8 _ZN3pag10PAGSurface11lockContextEb + 36
20  libpag                              0x000000010d4bc154 _ZN3pag10PAGSurface4drawEPNS_11RenderCacheENSt3__110shared_ptrINS_7GraphicEEEPNS_16BackendSemaphoreEb + 56
21  libpag                              0x000000010d4b3af8 _ZN3pag9PAGPlayer13flushInternalEPNS_16BackendSemaphoreE + 148
22  libpag                              0x000000010d4b3a20 _ZN3pag9PAGPlayer5flushEv + 140
23  libpag                              0x000000010d499778 _ZN3pag3PAG10SDKVersionEv + 10660
24  libpag                              0x000000010d497fbc _ZN3pag3PAG10SDKVersionEv + 4584
25  libpag                              0x000000010d498c84 _ZN3pag3PAG10SDKVersionEv + 7856
26  libpag                              0x000000010d4f8d88 _ZNK3pag11StrokeStyle6verifyEv + 9604
27  libpag                              0x000000010d4f911c _ZNK3pag11StrokeStyle6verifyEv + 10520
28  libpag                              0x000000010d4f9b38 _ZNK3pag11StrokeStyle6verifyEv + 13108
29  libsystem_pthread.dylib             0x00000001dc676338 _pthread_start + 116
30  libsystem_pthread.dylib             0x00000001dc674938 thread_start + 8

libc++abi: terminating with uncaught exception of type NSException

**Comment by domchen** [comment]
从有限的信息唯一可以确认的是堆栈信息是错的，因为几个函数之间没有调用关系。没有正确的符号看不出有效的信息。检查一下是否匹配错了符号的版本，或者是否在使用有问题的bugly还原符号，尝试用命令行手动还原一下。

**Comment by CodeJhF** [comment]
这条我们在当时的回复里已经给出结论，这里就不重复贴了，先关闭本条。
如果按当时的结论排查后仍有问题，欢迎回复补充最小复现 demo 与版本号（4.4 / 4.5），我们会重新看。（stale-close）
